source/appModules/totalcmd.py

```diff
@@ -25,25 +32,32 @@ def event_gainFocus(self):
 		if oldActivePannel !=self.windowControlID:
 			oldActivePannel=self.windowControlID
 			obj=self
+			obj2 = self
 			while obj and obj.parent and obj.parent.windowClassName!="TTOTAL_CMD":
 				obj=obj.parent
-			counter=0
-			while obj and obj.previous and obj.windowClassName!="TPanel":
+			while obj and obj.previous and obj.windowClassName!="Window":
 				obj=obj.previous
-				if obj.windowClassName!="TDrivePanel":
-					counter+=1
-			if counter==2:
-				ui.message(_("left"))
-			else:
-				ui.message(_("right"))
+			try:
+				if obj2.parent.parent.previous.firstChild.role  == 14:
```

Comment:
Can you please use the constant from controlTypes.py?
you can import controlTypes, and then do something like == controlTypes.ROLE_LIST